shared/src/business/entities/cases/RestrictedCase.ts

```diff
@@ -0,0 +1,26 @@
+import { Case } from '@shared/business/entities/cases/Case';
+import { JoiValidationEntity } from '@shared/business/entities/JoiValidationEntity';
+
+// An entity for case details for a case a user does not have access to
+export class RestrictedCase extends JoiValidationEntity {
+  public docketNumber?: string;
+  public docketNumberSuffix?: string;
+  public isPaper?: boolean;
+  public isSealed?: string;
+  public leadDocketNumber?: boolean;
+  public docketEntries: {}[];
```

Comment:
This is a compromise: we sometimes _expect_ "Case" data to have docketEntries. Rather than doing a falsy check in these situations, I thought it was cleaner to just make this an empty list. Now I'm having second thoughts, but I don't have strong feelings either way.